test/gtest/uct/ib/test_ib_md.cc

```diff
@@ -111,17 +114,22 @@ bool test_ib_md::has_ksm() const {
 }
 
 bool test_ib_md::check_umr(uct_ib_md_t *ib_md) const {
+    if (ib_md->dev.flags & UCT_IB_DEVICE_FLAG_MLX5_PRM) {
 #if HAVE_DEVX
-    return has_ksm();
+        if (ucs_derived_of(ib_md, uct_ib_mlx5_md_t)->flags & UCT_IB_MLX5_MD_FLAG_DEVX) {
+            return has_ksm();
+        } else {
+            return ib_md->relaxed_order;
```

Comment:
BTW, if we have relaxed_order, shouldn't we always have UMR, regardless of devx/exp situation?
for example even if has_ksm() would return false

Reply:
we should, but in current implementation DEVX && !KSM is exception

Reply:
i see, so this ok for v1.9 branch, but can you pls adjust this test here to have the more generic logic?

Reply:
you mean modify implementation for old FW?

Reply:
i mean make the test more independent of ib init logic
e.g if we have relaxed_order - we always expect 2nd key - so always return true

Reply:
but the test will fail than